docs/javascript-unit-tests.md

```diff
@@ -0,0 +1,712 @@
+[see the “Running tests” section](#running-tests) on this page.
+
+Now, let’s begin! Let’s make a new unit test fail,
+then we will make it pass, following the principles of TDD.
+
+1. Create 2 new files: a test file, and a JavaScript file.
+   Their names should match, with the test file adding a `-spec` suffix.
+   For example, if your script is named `my-javascript.js`,
+   then your test is named `my-javascript-spec.js`.
+1. Copy the code from this
+   [`sample-spec.js` file](https://github.com/cfpb/cfgov-refresh/blob/fc9ecc002f933a432322bd15f54be5eb2cffa6e3/test/unit_tests/js/modules/sample-spec.js)
+   into your own test file.
+1. Copy and paste from
+   [`sample.js`](https://github.com/cfpb/cfgov-refresh/blob/fc9ecc002f933a432322bd15f54be5eb2cffa6e3/cfgov/unprocessed/js/modules/sample.js)
+   into your new JavaScript file.
+
+
+### Folder structure (where to put your JavaScript and tests)
+
+JavaScript unit test files belong in the
+[`test/unit_tests/`](https://github.com/cfpb/cfgov-refresh/tree/master/test/unit_tests)
+directory.
+
+The folder structure of the test files mirrors
+the structure of the project JavaScript in
+[`cfgov/unprocessed/js/`](https://github.com/cfpb/cfgov-refresh/tree/master/cfgov/unprocessed/js).
+
+So my `sample-spec.js` file is in `test/unit_tests/modules/`
+and my `sample.js` file is in `cfgov/unprocessed/js/modules/`.
+I put these files in the `modules` folder because
+they are not atomic components, they are just sample files.
+Your JavaScript might be an atomic element,
+in which case it should go in the corresponding folder.
+If you're not sure where your JavaScript should go,
+[read about atomic components in cfgov-refresh](../atomic-structure/).
+
+!!! note "Child apps"
+    If you’re working on something in a child app,
+    put it in `test/unit_test/appname/js/`.
+    Otherwise, if you’re working on something that
+    belongs to cfgov-refresh generally,
+    it should go in the corresponding folder under `test/unit_test/js/`.
+
+Now that you have your sample JS and test files in the right places,
+let’s try running them and see what happens!
+I’ll refer to `sample-spec.js` and `sample.js` in the instructions below,
+but you should work in your own new test file and JavaScript file
+to save and commit your changes.
+
+1. Edit line 6 of your spec file and remove the `.skip` method.
+   The line should now read:
+   ```js
+   it( 'should return a string with expected value', () => {
+     …
+   } );
+   ```
+1. Run your sample test using `gulp test:unit --specs=js/modules/sample-spec.js`
+   (substituting your own filename).
+   The test should fail – this is expected.
+   Remember, when doing TDD, we want to write our test to fail first,
+   then write the corresponding JavaScript that will make the test pass.
+1. Make the test pass by changing your script’s line 7
+   ([see `sample.js`](https://github.com/cfpb/cfgov-refresh/blob/master/cfgov/unprocessed/js/modules/sample.js))
```

Comment:
Maybe link to the exact SHA as you did [above](https://github.com/cfpb/cfgov-refresh/pull/4727/files#diff-5b5157ceb64552e1fba15c850930a0afR84) just in case this file ever changes?